subworkflows/local/input_check.nf

```diff
@@ -2,40 +2,39 @@
 // Check input samplesheet and get read channels
 //
 
-include { SAMPLESHEET_CHECK } from '../../modules/local/samplesheet_check'
+//
+// Include plugins
+//
+include { fromSamplesheet                          } from 'plugin/nf-validation'
 
 workflow INPUT_CHECK {
     take:
     samplesheet // file: /path/to/samplesheet.csv
 
     main:
-    SAMPLESHEET_CHECK ( samplesheet )
-        .csv
-        .splitCsv ( header:true, sep:',' )
-        .map { create_fastq_channel(it) }
+    Channel.fromSamplesheet("input")
```

Comment:
This is working fine, so my comment is only a suggestion. You can remove the whole `input_check` subworkflow by creating the input channel in the main script. If you want to see an example, rnaseq has a [PR](https://github.com/nf-core/rnaseq/pull/1058) with these changes.

Reply:
I tried but I have trouble in appending the meta.id by map with pseudo-code:
```
.map{
   meta, fq1, fq2 ->
      meta.id = meta.group + "_REP" + meta.replicate + "_T" + meta.techniquerep // I have trouble here
      [meta, [fq1, fq2]]
}
```

Reply:
This is modifying the meta map in place, which can lead to errors in the pipeline. An alternative is using the operators `+` or `-` which create a copy of the object instead.
`meta - meta.subMap('id') + [id: meta.group + "_REP" + meta.replicate + "_T" + meta.techniquerep`

Reply:
I removed the input_check.nf. Could you please take a second look?